### PR TITLE
Use `dataclass_transform` instead of our custom one

### PR DESCRIPTION
**Pull request by patrick91**
I hope this improves support for other type-checkers, like Pyre.

### CODE REVIEW

**Comment by botberry** [comment]
# Pre-release
:wave:

Releasing commit [8b7ba401a64b5ad94e8ee5e7042b63408c7d084c] to PyPi as pre-release! :package:

**Comment by botberry** [comment]
# Pre-release
:wave:

Pre-release **0.134.0.dev.1664917721** [8b7ba401a64b5ad94e8ee5e7042b63408c7d084c] has been released on PyPi! :rocket:
You can try it by doing:

```shell
poetry add strawberry-graphql==0.134.0.dev.1664917721
```

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release removes our custom `__dataclass_transfor__` decorator
and replaces it with typing-extension's one.

... todo add more text


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @patrick91 for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Release type: minor
+
+This release removes our custom `__dataclass_transfor__` decorator
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release removes our custom `__dataclass_transform__` decorator
```

**Comment by botberry** [comment]
## Apollo Federation Subgraph Compatibility Results

<table>
<thead>
<tr><th>Federation 1 Support</th><th>Federation 2 Support</th></tr>
</thead>
<tbody>
<tr><td><table><tr><th><code>_service</code></th><td>🟢</td></tr><tr><th><code>@key (single)</code></th><td>🟢</td></tr><tr><th><code>@key (multi)</code></th><td>🟢</td></tr><tr><th><code>@key (composite)</code></th><td>🟢</td></tr><tr><th><code>repeatable @key</code></th><td>🟢</td></tr><tr><th><code>@requires</code></th><td>🟢</td></tr><tr><th><code>@provides</code></th><td>🟢</td></tr><tr><th><code>federated tracing</code></th><td>🔲</td></tr></table></td><td><table><tr><th><code>@link</code></th><td>🟢</td></tr><tr><th><code>@shareable</code></th><td>🟢</td></tr><tr><th><code>@tag</code></th><td>🟢</td></tr><tr><th><code>@override</code></th><td>🟢</td></tr><tr><th><code>@inaccessible</code></th><td>🟢</td></tr><tr><th><code>@composeDirective</code></th><td>🟢</td></tr><tr><th><code>@interfaceObject</code></th><td>🟢</td></tr></table></td></tr>
</tbody>
</table>

Learn more:
* [Apollo Federation Subgraph Specification](https://www.apollographql.com/docs/federation/subgraph-spec/)
* [Compatibility Tests](https://github.com/apollographql/apollo-federation-subgraph-compatibility/blob/main/COMPATIBILITY.md)

**Pull request merged**